docs/source/pagination/introduction.mdx

```diff
@@ -0,0 +1,127 @@
+
+The `previousPages` field contains all the pages that have been fetched in the reverse direction, and exposes the paginated query's `Data` type. The `initialPage` field contains the first page of data that was fetched, and is an instance of the initial query's `Data` type. The `nextPages` field contains all the pages that have been fetched in the forward direction, and is a list of the paginated query's `Data` type.
+
+If you want to use a custom response type, you can provide a `transform` closure when initializing the `GraphQLQueryPager` instance. This closure is called whenever the pager fetches a new page of data, and is used to transform the raw response into the desired response type. 
+
+```swift
+struct Person {
+    let id: String
+    let name: String
+}
+
+let pager = GraphQLQueryPager.makeForwardCursorQueryPager(
+    client: apolloClient,
+    queryProvider: { page in
+        MyQuery(first: 10, after: page?.cursor ?? .none)
+    },
+    extractPageInfo: { data in
+        CursorBasedPagination.Forward(
+            hasNext: data.values.pageInfo.hasNextPage ?? false,
+            endCursor: data.values.pageInfo.endCursor
+        )
+    },
+    transform: { data in 
+        data.values.nodes.map { Person(id: $0.id, name: $0.name) }
+    }
+)
+```
+
+In this example, the `GraphQLQueryPager` instance is initialized with a `transform` closure that transforms the raw response into an array of `Person` objects. Alternatively, we can make use of the `GraphQLQueryPager`'s `map` method to achieve the same result:
+
+```swift
+pager.map { data in 
+    data.values.nodes.map { Person(id: $0.id, name: $0.name) }
+}.sink { result in
+    switch result {
+    case .success(let people):
+        // Handle the people
+    case .failure(let error):
+        // Handle the error
+    }
+}
+```
+
+## Fetching Data
+
+The `GraphQLQueryPager` class provides several methods to fetch paginated data: `fetch`, `refetch`, `loadNext`, `loadPrevious`, and `loadAll`. 
+
+The `fetch` method is used to fetch the first page of data and must be called before `loadNext` or `loadPrevious` can be called. The `refetch` method is used to refetch the first page of data, and additionally resets the pager to its initial state. The `loadNext` method is used to fetch the next page of data, and the `loadPrevious` method is used to fetch the previous page of data. The `loadAll` method is used to fetch all pages of data. 
+
+## Cancelling a Pager
+
+The `GraphQLQueryPager` class provides a `cancel` method, which can be used to cancel any ongoing fetch operations as well as cease any watcher subscriptions. Once cancelled, the pager's state is reset. Any subscriber that was subscribed to the pager will remain subscribed and will continue to receive updates. 
+
+## Error handling
+
+Network errors are exposed to the caller via the `Result` type. Usage errors are thrown as `Error` types or exposed as callbacks in each fetch method. 
```

Comment:
I know what you are talking about here because we discussed it. But I think this needs a little more detail.

Clarity on the difference between network errors and usage errors as well as maybe a code snippet showing where they show up.